### ISSUE TITLE
Pact mock server cannot start 

### ISSUE THREAD

**Issue by toannh0104** [pact-foundation/pact-js-core#409]
I'm setup pact test and saw this issue 
```
[2022-10-03 09:41:28.784 +0000] DEBUG (16466 on M01GMBGH0ADMD6Q): pact-node@10.12.2: Created 'standalone/darwin-1.88.49/pact/bin/pact-mock-service' process with PID: 16479
[2022-10-03 09:41:29.383 +0000] DEBUG (16466 on M01GMBGH0ADMD6Q): pact-node@10.12.2: INFO  WEBrick 1.3.1
INFO  ruby 2.2.2 (2015-04-13) [x86_64-darwin13]

[2022-10-03 09:41:29.384 +0000] DEBUG (16466 on M01GMBGH0ADMD6Q): pact-node@10.12.2: INFO  WEBrick::HTTPServer#start: pid=16479 port=8081

[2022-10-03 09:41:29.429 +0000] TRACE (16466 on M01GMBGH0ADMD6Q): pact@9.15.5: outgoing request
[2022-10-03 09:41:29.431 +0000] TRACE (16466 on M01GMBGH0ADMD6Q): pact@9.15.5: 
    body: "<HTML><HEAD>\r\n        <TITLE>Access Denied</TITLE>\r\n        </HEAD>\r\n        <BODY>\r\n        <FONT face=\"Helvetica\">\r\n        <big><strong></strong></big><BR>\r\n        </FONT>\r\n        <blockquote>\r\n        <TABLE border=0 cellPadding=1 width=\"80%\">\r\n        <TR><TD>\r\n        <FONT face=\"Helvetica\">\r\n        <big>Access Denied (authentication_failed)</big>\r\n        <BR>\r\n        <BR>\r\n        </FONT>\r\n        </TD></TR>\r\n        <TR><TD>\r\n        <FONT face=\"Helvetica\">\r\n        Your credentials could not be authenticated: \"General authentication failure due to bad user ID or authentication token.\". You will not be permitted access until your credentials can be verified.\r\n        </FONT>\r\n        </TD></TR>\r\n        <TR><TD>\r\n        <FONT face=\"Helvetica\">\r\n        This is typically caused by an incorrect username and/or password, but could also be caused by network problems.\r\n        </FONT>\r\n        </TD></TR>\r\n        <TR><TD>\r\n        <FONT face=\"Helvetica\" SIZE=2>\r\n        <BR>\r\n        For assistance, contact helpdesk.\r\n        </FONT>\r\n        </TD></TR>\r\n        </TABLE>\r\n        <FONT face=\"Helvetica\" SIZE=2>\r\n        <p>Client URL : http:&#x2F;&#x2F;127.0.0.1:8081&#x2F;</p>\r\n        <p>Client IP Address : 10.9.116.79</p>\r\n        <p>Client Username : toann</p>\r\n        <p>Date & Time : [03&#x2F;Oct&#x2F;2022:17:41:29 +0800]</p>\r\n        <p>Reference Device : n01vbcproxy11</p>\r\n        </blockquote>\r\n        </FONT>\r\n        </BODY></HTML>\r\n        "
    headers: {
      "proxy-authenticate": "NEGOTIATE, NTLM, BASIC realm=\"TEST_PROD\"",
      "cache-control": "no-cache",
      "pragma": "no-cache",
      "content-type": "text/html; charset=utf-8",
      "proxy-connection": "close",
      "set-cookie": [
        "BCSI-CS-2398159d1db5bcba=2; Path=/"
      ],
      "connection": "close",
      "content-length": "1483"
    }
    statusCode: 407
[2022-10-03 09:41:30.378 +0000] TRACE (16466 on M01GMBGH0ADMD6Q): pact@9.15.5: outgoing request
[2022-10-03 09:41:30.379 +0000] TRACE (16466 on M01GMBGH0ADMD6Q): pact@9.15.5: 
```

It seems some proxy is required, could you let me know how to do proxy config on setup pact test?

**Comment by mefellows** [comment]
Yes, definitely looks like a proxy issue. Have you set the appropriate proxy headers e.g. `http_proxy`, `https_proxy` and `no_proxy`?

**Comment by toannh0104** [comment]
How can I do those setup on macos?, I didn't find any documents for http proxy setup on pact's documents

**Comment by mefellows** [comment]
https://github.com/pact-foundation/pact-js/blob/e3607bd3ee4d6f45c4c0813e75676075635780d1/docs/troubleshooting.md#corporate-proxies--firewalls

TL;DR - it's the usual proxy environment variables.

**Comment by toannh0104** [comment]
I run `no_proxy=localhost,127.0.0.1,$no_proxy export no_proxy` then run the `npm run test:pact` but got same error

**Comment by toannh0104** [comment]
I no idea but the issue is resolved by use host is 'localhost' instead of '127.0.0.1' or default value
